metaserver/src/main/scala/scala/meta/languageserver/search/InMemorySymbolIndex.scala

```diff
@@ -185,4 +188,38 @@ class InMemorySymbolIndex(
     Effects.IndexSemanticdb
   }
 
+  private def hasOneOfFlag(flags: Long, flag: Long): Boolean =
+    (flags & flag) != 0L
+
+  override def workspaceSymbols(query: String): List[SymbolInformation] = {
+    import scala.meta.semanticdb._
+    import scala.meta.languageserver.ScalametaEnrichments._
+    val result = symbolIndexer.allSymbols.toIterator
+      .withFilter { symbol =>
+        symbol.definition.isDefined && symbol.definition.get.uri
+          .startsWith("file:")
+      }
+      .collect {
+        case i.SymbolData(sym, Some(pos), _, flags, name, _)
+            if hasOneOfFlag(flags, CLASS | TRAIT | OBJECT) && {
+              // NOTE(olafur) fuzzy-wuzzy doesn't seem to do a great job
+              // for camelcase searches like "DocSymPr" when looking for
+              // "DocumentSymbolProvider. We should try and port something
+              // like https://blog.forrestthewoods.com/reverse-engineering-sublime-text-s-fuzzy-match-4cffeed33fdb
+              // instead.
+              FuzzySearch.partialRatio(query, name) >= 90
+            } =>
+          val symbolKind =
```

Comment:
Can you move this logic to private method for consistency to other places where we do `A -> SymbolKind` conversions?

Reply:
Good idea. I created a new `org.langmeta.semanticdb.SemanticdbEnrichments` to access the package private `HasFlags` trait